src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -47,6 +48,25 @@ public Lecture filterActiveAttachments(Lecture lectureWithAttachments, User user
         return lectureWithAttachments;
     }
 
+    /**
+     * Returns lecture with only active lectureUnits
+     *
+     * @param lectureWithLectureUnits lecture that has lectureUnits
+     * @return lecture with filtered lectureUnits
+     */
+    public Lecture filterActiveLectureUnits(Lecture lectureWithLectureUnits) {
```

Comment:
Why not directly filter the lecture units in the SQL-querry?

Reply:
First the idea was to allow tutors to be able to retrieve the units and therefore i would filter only for the Students. But now i do only show active units,  only if unit is released the users will be able to reference slides. So i moved the filter directly to sql query now.